common/src/main/java/org/apache/celeborn/common/network/server/BufferStreamManager.java

```diff
@@ -313,74 +321,118 @@ public synchronized void setupDataPartitionReader(
               streamId,
               channel,
               () -> recycleStream(streamId));
-      // allocate resources when the first reader is registered
-      boolean allocateResources = readers.isEmpty();
-      readers.add(dataPartitionReader);
-      streamReaders.put(streamId, dataPartitionReader);
+      readers.put(streamId, dataPartitionReader);
 
-      // create initial buffers for read
-      if (allocateResources && buffers == null) {
+      // allocate resources when the first reader is registered
+      if (!bufferQueueInitialized) {
         memoryManager.requestReadBuffers(
-            minReadBuffers,
-            maxReadBuffers,
-            fileInfo.getBufferSize(),
-            (allocatedBuffers, throwable) ->
-                MapDataPartition.this.onBuffer(new LinkedBlockingDeque<>(allocatedBuffers)));
+            new ReadBufferRequest(
+                minReadBuffers,
```

Comment:
In the default config, minReadBuffers is 16m and maxReadBuffers is 32m, which is bad for cases where the file size is smaller than 16m or 32m. IMO we need to add a fileSize in FileInfo, and ensure minBuffers and maxBuffers not exceed the fileSize.